Lib/test/test_zipfile/test_core.py

```diff
@@ -3447,6 +3447,62 @@ def test_too_short(self):
         self.assertEqual(
             b"zzz", zipfile._Extra.strip(b"zzz", (self.ZIP64_EXTRA,)))
 
+class StoredZipExtFileRandomReadTest(unittest.TestCase):
+    def test_random_read(self):
+        from _pyio import BytesIO
+        class StatIO(BytesIO):
+            def __init__(self):
+                super().__init__()
+                self.bytes_read = 0
+
+            def read(self, size=-1):
+                bs = super().read(size)
+                self.bytes_read += len(bs)
+                return bs
+
+        sio = StatIO()
+        # 20000 bytes
+        txt = b'0123456789' * 2000
+
+        # seek length must be greater than ZipExtFile.MIN_READ_SIZE (4096)
+        min_size = zipfile.ZipExtFile.MIN_READ_SIZE
+        self.assertGreaterEqual(min_size, 100)
+
+        with zipfile.ZipFile(sio, "w", compression=zipfile.ZIP_STORED) as zipf:
+            zipf.writestr("foo.txt", txt)
+
+        # check random seek and read on a file
+        with zipfile.ZipFile(sio, "r") as zipf:
+            with zipf.open("foo.txt", "r") as fp:
+                # forward seek
+                old_count = sio.bytes_read
+                fp.seek(10002, os.SEEK_CUR)
+                self.assertEqual(fp.tell(), 10002)
+                self.assertEqual(fp._left, fp._compress_left)
+                arr = fp.read(100)
+                self.assertEqual(fp.tell(), 10102)
+                self.assertEqual(arr, txt[10002:10102])
+                self.assertEqual(fp._left, fp._compress_left)
+                d = sio.bytes_read - old_count
+                self.assertLessEqual(d, min_size)
```

Comment:
I had to think about why we were doing this and a comment would here would help. Feel free to change this if this wasn't the intent of this test :)

```suggestion
                # Test this optimized read hasn't rewound and read from the
                # start of the file (as in the case of the unoptimized path)
                d = sio.bytes_read - old_count
                self.assertLessEqual(d, min_size)
```